src/ansys/tools/visualization_interface/backends/plotly/plotly_interface.py

```diff
@@ -154,27 +155,34 @@ def plot(
             The object to plot. Can be a PyVista PolyData, MultiBlock, a MeshObjectPlot, or a Plotly Mesh3d.
         plotting_options : dict
             Additional plotting options.
+        name : str, optional
+            Name of the mesh for labeling in Plotly. Overrides the name from MeshObjectPlot if provided.
         """
         if isinstance(plottable_object, MeshObjectPlot):
             mesh = plottable_object.mesh
+            name = plottable_object.name if name is None else name
         else:
             mesh = plottable_object
 
         if isinstance(mesh, (PolyData, pv.MultiBlock)):
             mesh_result = self._pv_to_mesh3d(mesh)
-
             # Handle both single mesh and list of meshes
             if isinstance(mesh_result, list):
                 # MultiBlock case - add all meshes
                 for mesh_3d in mesh_result:
+                    mesh_3d.name = name if name is not None else mesh_3d.name
```

Comment:
Same suggestion here:


```suggestion
                    mesh_3d.name = name or mesh_3d.name
```